src/runtime/Temporal.h

```diff
@@ -37,9 +39,49 @@ class Temporal : public Object {
         return true;
     }
 
-    bool hasInitializedTemporalDate() const
+    static Value toTemporalOverflow(ExecutionState& state, const Value& normalizedOptions)
     {
-        return false;
+        if (normalizedOptions.isUndefined()) {
+            return state.context()->staticStrings().lazyConstrain().string();
+        }
+        Value matcherValues[2] = { state.context()->staticStrings().lazyConstrain().string(), state.context()->staticStrings().lazyReject().string() };
+        return Intl::getOption(state, normalizedOptions.asObject(), state.context()->staticStrings().lazyoverflow().string(), Intl::StringValue, matcherValues, 2, matcherValues[0]);
+    }
+
+    static Value prepareTemporalFields(ExecutionState& state, const Value& fields, const ValueVector& fieldNames, const ValueVector& requiredFields)
+    {
+        ASSERT(fields.isObject());
+        auto* result = new Object(state);
+        std::string table13[14] = { "year", "month", "monthCode", "day", "hour", "minute", "second", "millisecond",
+                                    "microsecond", "nanosecond", "offset", "era", "eraYear", "timeZone" };
+        for (auto property : fieldNames) {
+            Value value;
+            fields.asObject()->get(state, ObjectPropertyName(state, fields), property).value(state, value);
+            if (value.isUndefined()) {
+                if (std::find(requiredFields.begin(), requiredFields.end(), property) != requiredFields.end()) {
```

Comment:
In this for loop, `value` is always set as undefined.
Is this intended?

Reply:
At the end of line `59` it's set to a value or i'm misunderstanding the usage of `object->get` and `.value` functions?

Reply:
No, `Object::get` returns a `ObjectGetResult` and `ObjectGetResult::value` returns the value that you (might) want to get.